src/core/observer/index.js

```diff
@@ -188,6 +188,13 @@ export function defineReactive (
  * already exist.
  */
 export function set (obj: Array<any> | Object, key: any, val: any) {
+  if (arguments.length < 3) {
+    process.env.NODE_ENV !== 'production' && warn(
+      'Set now requires you to specifiy the object you are modifying' +
```

Comment:
I think it would be better to explicitly say what we talk about. `Set` is also a type in Javascript.
    Vue.set() / this.$set() now requires ...